sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -83,17 +84,41 @@ export class ClientContext {
     if (cosmosClientOptions.aadCredentials) {
       this.pipeline = createEmptyPipeline();
       const hrefEndpoint = sanitizeEndpoint(cosmosClientOptions.endpoint);
-      const scope = `${hrefEndpoint}/.default`;
+
+      // Use custom AAD scope if provided, otherwise use account-based scope
+      const accountScope = `${hrefEndpoint}/.default`;
+      const primaryScope = cosmosClientOptions.aadScope || accountScope;
+      const fallbackScope = AAD_DEFAULT_SCOPE;
+      const AUTH_PREFIX = `type=aad&ver=1.0&sig=`;
+
       this.pipeline.addPolicy(
         bearerTokenAuthenticationPolicy({
           credential: cosmosClientOptions.aadCredentials,
-          scopes: scope,
+          scopes: primaryScope,
           challengeCallbacks: {
             async authorizeRequest({ request, getAccessToken }) {
-              const tokenResponse = await getAccessToken([scope], {});
-              const AUTH_PREFIX = `type=aad&ver=1.0&sig=`;
-              const authorizationToken = `${AUTH_PREFIX}${tokenResponse.token}`;
-              request.headers.set("Authorization", authorizationToken);
+              try {
+                const tokenResponse = await getAccessToken([primaryScope], {});
+
+                const authorizationToken = `${AUTH_PREFIX}${tokenResponse.token}`;
+                request.headers.set("Authorization", authorizationToken);
+              } catch (error: any) {
+                // If no custom scope is provided and we get AADSTS500011 error,
+                // fallback to the default Cosmos scope
+                if (!cosmosClientOptions.aadScope && error?.message?.includes("AADSTS500011")) {
```

Comment:
AADSTS500011  is in error message/string? do we have some status codes on which we can rely on?

Reply:
It is part of error string only.

Reply:
move this error code to constants file

Reply:
moved code to constants, we can take a call on diagnostics changes in next version.

Reply:
For now, I have added a boolean value in diagnostics clientConfig which is true if custom scope is provided during client initialization. I am not sure curretnly if there is any way to track in diagnostics if fallback scope gets used.